src/managed/Microsoft.Extensions.DependencyModel/StreamBufferWriter.Netstandard2_0.cs

```diff
@@ -0,0 +1,92 @@
+using System.IO;
+
+namespace Microsoft.Extensions.DependencyModel
+{
+    internal struct StreamBufferWriter : IBufferWriter<byte>, IDisposable
+    {
+        private Stream _stream;
+        private byte[] _rentedBuffer;
+
+        private const int MinimumBufferSize = 256;
+
+        public StreamBufferWriter(Stream stream, int bufferSize = MinimumBufferSize)
+        {
+            if (bufferSize <= 0)
+                throw new ArgumentException(nameof(bufferSize));
+
+            _stream = stream ?? throw new ArgumentNullException(nameof(stream));
+
+            _rentedBuffer = ArrayPool<byte>.Shared.Rent(bufferSize);
+        }
+
+        public void Advance(int count)
+        {
+            _stream.Write(_rentedBuffer, 0, count);
+            _rentedBuffer.AsSpan(0, count).Clear();
+        }
+
+        /// <summary>
+        /// Returns rented buffer back to the pool
+        /// </summary>
+        public void Dispose()
+        {
+            _rentedBuffer.AsSpan().Clear();
+            ArrayPool<byte>.Shared.Return(_rentedBuffer);
+            _rentedBuffer = null;
+        }
+
+        public Memory<byte> GetMemory(int sizeHint = 0)
+        {
+            if (sizeHint < 0)
+                throw new ArgumentException(nameof(sizeHint));
+
+            if (sizeHint == 0)
+            {
+                int newSize = _rentedBuffer.Length == 0 ? MinimumBufferSize : checked(_rentedBuffer.Length * 2);
+                byte[] temp = _rentedBuffer;
+                _rentedBuffer = ArrayPool<byte>.Shared.Rent(newSize);
+                temp.AsSpan().Clear();
+                ArrayPool<byte>.Shared.Return(temp);
+            }
+            else if (sizeHint > _rentedBuffer.Length)
+            {
+                var temp = _rentedBuffer;
+                _rentedBuffer = ArrayPool<byte>.Shared.Rent(sizeHint);
+                temp.AsSpan().Clear();
+                ArrayPool<byte>.Shared.Return(temp);
+            }
+            Debug.Assert(_rentedBuffer.Length > 0);
+            return _rentedBuffer;
+        }
+
+        public Span<byte> GetSpan(int sizeHint = 0)
+        {
+            if (sizeHint < 0)
+                throw new ArgumentException(nameof(sizeHint));
+
+            if (sizeHint == 0)
+            {
+                int newSize = _rentedBuffer.Length == 0 ? MinimumBufferSize : checked(_rentedBuffer.Length * 2);
+                byte[] temp = _rentedBuffer;
+                _rentedBuffer = ArrayPool<byte>.Shared.Rent(newSize);
+                temp.AsSpan().Clear();
+                ArrayPool<byte>.Shared.Return(temp);
+            }
+            else if (sizeHint > _rentedBuffer.Length)
+            {
+                var temp = _rentedBuffer;
+                _rentedBuffer = ArrayPool<byte>.Shared.Rent(sizeHint);
+                temp.AsSpan().Clear();
+                ArrayPool<byte>.Shared.Return(temp);
+            }
+            Debug.Assert(_rentedBuffer.Length > 0);
+            return _rentedBuffer;
```

Comment:
Rather than duplicating the whole method, can't this just be:
```C#
public Span<byte> GetSpan(int sizeHint = 0) => GetMemory(sizeHint).Span;
```
?  Is the extra overhead of going through the `Memory<T>` prohibitive?  And if it is, could you just refactor all of this into a private helper that returns the array, and then wrap GetMemory and GetSpan around that?